client/network/src/discovery.rs

```diff
@@ -386,37 +378,27 @@ impl DiscoveryBehaviour {
 	///
 	/// Identifies Kademlia instances by their [`ProtocolId`] and kbuckets by the base 2 logarithm
 	/// of their lower bound.
-	pub fn num_entries_per_kbucket(
-		&mut self,
-	) -> impl ExactSizeIterator<Item = (&ProtocolId, Vec<(u32, usize)>)> {
-		self.kademlias.iter_mut().map(|(id, kad)| {
-			let buckets = kad
-				.kbuckets()
+	pub fn num_entries_per_kbucket(&mut self) -> Option<Vec<(u32, usize)>> {
```

Comment:
libp2p Kademlia behavior functions used here all take `&mut self`, so the only option is to call `as_mut()`. May be this is because DHT lookups modify the internal state.